torchgeo/datasets/sentinel.py

```diff
@@ -25,6 +25,189 @@ class Sentinel(RasterDataset):
+    # l:          Entire Area (e) or Clipped Area (c)
+    # m:          Dead Reckoning (d) or DEM Matching (m)
+    # ssss:       Product ID
+    filename_glob = "S1*{}.*"
+    filename_regex = r"""
+        ^S1(?P<mission>[AB])
+        _(?P<mode>SM|IW|EW|WV)
+        _(?P<date>\d{8}T\d{6})
+        _(?P<polarization>[DS][HV])
+        (?P<orbit>[PRO])
+        _RTC(?P<spacing>\d{2})
+        _(?P<package>G)
+        _(?P<output1>[gs])
+        (?P<output2>[pda])
+        (?P<mask>[uw])
+        (?P<filter>[nf])
+        (?P<area>[ec])
+        (?P<matching>[dm])
+        _(?P<product>[0-9A-Z]{4})
+        _(?P<band>[VH]{2})
+        \.
+    """
+    date_format = "%Y%m%dT%H%M%S"
+    all_bands = ["HH", "HV", "VV", "VH"]
+    separate_files = True
+
+    def __init__(
+        self,
+        root: str = "data",
+        crs: Optional[CRS] = None,
+        res: float = 10,
+        bands: Sequence[str] = ["VV", "VH"],
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        cache: bool = True,
+    ) -> None:
+        """Initialize a new Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            bands: bands to return (defaults to all bands)
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+
+        Raises:
+            AssertionError: if ``bands`` is invalid
+            FileNotFoundError: if no files are found in ``root``
+        """
+        # Sentinel-1 uses dual polarization, it can only transmit either
+        # horizontal or vertical at a single time
+        assert list(bands) == ["HH", "HV"] or list(bands) == ["VV", "VH"]
+
+        self.filename_glob = self.filename_glob.format(bands[0])
+
+        super().__init__(root, crs, res, bands, transforms, cache)
+
+    def plot(
+        self,
+        sample: Dict[str, Any],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`RasterDataset.__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        # Either HH/HV or VV/VH, doesn't really matter which
+        hh = sample["image"][0]
+        hv = sample["image"][1]
+        hh_hv = hh / hv
+
+        # https://gis.stackexchange.com/a/400780/123758
+        hh = torch.clamp(hh / 0.3, min=0, max=1)
```

Comment:
these variable names are confusing as "hh" could mean the "HH" band or the "VH" band -- maybe using variables like `horizontal_receive` and `vertical_receive` would be better?

Reply:
hh means HH or VV, not HH or VH.

* hh: the band where transmit == receive
* hv: the band where transmite != receive

Wasn't sure what a better name would be. It doesn't matter too much since it's a local variable that users can't access. But I would at least like to make this clear. Will clarify this comment.